### PR TITLE
Lay initial groundwork for file configuration

### PR DESCRIPTION
**Pull request by jack-berg**
Initial followup to [OTEP #225](https://github.com/open-telemetry/oteps/pull/225).

This lays the initial groundwork for file configuration in the specification. There are placeholders TODOs for many various followup tasks. Additional work is also needed to figure out how to merge [SDK Configuration](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/sdk-configuration.md), [SDK Environment Variables](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/sdk-environment-variables.md) and file configuration into a cohesive story.

This PR proposes the configuration schema live in `opentelemetry-specification/specification/configuration/schema/`, which implies that at some point additional build steps will be needed to confirm changes to the schema are valid and allowed.

cc @MrAlias, @codeboten

### CODE REVIEW

**Review by MrAlias — APPROVED** [review]

**Review by codeboten — APPROVED** [review]

**Review by trask — APPROVED** [review]

File: specification/configuration/file-configuration.md [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+
+## Configuration Schema
+
+The configuration schema is defined
+in [./schema/](./schema/opentelemetry_configuration.json)
+using [JSON Schema](https://json-schema.org/).
+
+### Stability Definition
+
+TODO: define stability guarantees and backwards compatibility
```
Comment (bogdandrutu) [inline specification/configuration/file-configuration.md:39]: Is this not already defined at project level? Why do we need a separate thing?

File: specification/configuration/file-configuration.md [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+TODO: define acceptable file formats
+
+TODO: define environment variable substitution
+
+## SDK Configuration
+
+SDK configuration defines the interfaces and operations that SDKs are expected
+to expose to enable file based configuration.
+
+### Configuration Model
```
Comment (bogdandrutu) [inline specification/configuration/file-configuration.md:55]: A bit confused about schema/model/file and order of defining them. Also for names, I would expect the current schema to be "data model", current model to be "in memory"

File: specification/configuration/file-configuration.md (hunk 1 above)
Comment (bogdandrutu) [reply specification/configuration/file-configuration.md:39]: But that is a separate repo, here we already have rules in this repo.... Does that mean we should have a separate repo?

File: specification/configuration/file-configuration.md (hunk 2 above)
Comment (jack-berg) [reply specification/configuration/file-configuration.md:55]: That seems reasonable. Could change `Configuration schema` to `Configuration model`, and say that the model is defined by the JSON schema.

File: specification/configuration/file-configuration.md (hunk 1 above)
Comment (jack-berg) [reply specification/configuration/file-configuration.md:39]: We could consider having a separate repo. The rules will definitely need to be specially considered for JSON schema. JSON schema supports all sorts of syntax and it's not always obvious / intuitive whether a particular change would be breaking or not. I suspect we'll need to learn as we go.

File: specification/configuration/schema/opentelemetry_configuration.json [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+    "additionalProperties": false,
+    "properties": {
+        "file_format": {
+            "type": "number"
+        }
+    },
+    "required": [
+        "file_format"
+    ]
+}
```
Comment (codeboten) [inline specification/configuration/schema/opentelemetry_configuration.json:15]:
```suggestion
}

```

**Review by carlosalberto — APPROVED** [review]

File: specification/configuration/file-configuration.md [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+
+## SDK Configuration
+
+SDK configuration defines the interfaces and operations that SDKs are expected
+to expose to enable file based configuration.
+
+### In-Memory Configuration Model
+
+SDKs SHOULD define an in-memory representation of
+the [Configuration Model](#configuration-model) called `Configuration`.
```
[the remark this answers is not in the document]
Comment (jack-berg) [reply specification/configuration/file-configuration.md:58]: The advantage of exporing a configuration model is that we're able to support more complex configuration scenarios without providing implementations for those. For example, suppose a user wants to merge multiple configuration files. We (opentelemetry) don't want to deal with defining nuanced merge semantics which are bound to cause to disagreements, but a user can still  parse multiple config files to corresponding configuration models, and define their own merge function.

If this type of thing is possible with a language's idiomatic configuration tooling, then perhaps its unnecessary to expose a configuration model.

File: specification/configuration/file-configuration.md (hunk 1 above)
Comment (jack-berg) [reply specification/configuration/file-configuration.md:39]: As discussed in the 4/11/23 spec SIG, I've opened #3379 to discuss breaking out the schema into a separate repo. Let's keep it as is for the purposes of this PR.

File: specification/configuration/file-configuration.md (hunk 4 above)
[the remark this answers is not in the document]
Comment (MrAlias) [reply specification/configuration/file-configuration.md:58]: > Got it, I suggest that we reword this to "SDKs SHOULD provide an in-memory representation of the Configuration Model. In general the SDKs are encouraged to follow their language idiomatic way, in case an SDK has to invent its own way, the name `Configuration` is recommended".

LGTM with the caveat that the last recommendation should be made normative with "RECOMMENDED".

**Review by reyang — APPROVED** [review]
